src/api/l_audio.c

```diff
@@ -79,170 +97,96 @@ static int l_lovrAudioGetPose(lua_State* L) {
   return 7;
 }
 
-static int l_lovrAudioGetPosition(lua_State* L) {
-  float position[4];
-  lovrAudioGetPosition(position);
-  lua_pushnumber(L, position[0]);
-  lua_pushnumber(L, position[1]);
-  lua_pushnumber(L, position[2]);
-  return 3;
-}
-
-static int l_lovrAudioGetVelocity(lua_State* L) {
-  float velocity[4];
-  lovrAudioGetVelocity(velocity);
-  lua_pushnumber(L, velocity[0]);
-  lua_pushnumber(L, velocity[1]);
-  lua_pushnumber(L, velocity[2]);
-  return 3;
-}
-
-static int l_lovrAudioGetVolume(lua_State* L) {
-  lua_pushnumber(L, lovrAudioGetVolume());
-  return 1;
+static int l_lovrAudioSetPose(lua_State *L) {
+  int index = 1;
+  float position[4], orientation[4];
+  index = luax_readvec3(L, index, position, NULL);
+  index = luax_readquat(L, index, orientation, NULL);
+  lovrAudioSetPose(position, orientation);
+  return 0;
 }
 
-static int l_lovrAudioIsSpatialized(lua_State* L) {
-  lua_pushboolean(L, lovrAudioIsSpatialized());
+static int l_lovrAudioGetCaptureStream(lua_State* L) {
+  SoundData* soundData = lovrAudioGetCaptureStream();
+  luax_pushtype(L, SoundData, soundData);
   return 1;
 }
 
-static int l_lovrAudioNewMicrophone(lua_State* L) {
-  const char* name = luaL_optstring(L, 1, NULL);
-  int samples = luaL_optinteger(L, 2, 1024);
-  int sampleRate = luaL_optinteger(L, 3, 8000);
-  int bitDepth = luaL_optinteger(L, 4, 16);
-  int channelCount = luaL_optinteger(L, 5, 1);
-  Microphone* microphone = lovrMicrophoneCreate(name, samples, sampleRate, bitDepth, channelCount);
-  luax_pushtype(L, Microphone, microphone);
-  lovrRelease(Microphone, microphone);
+static int l_lovrAudioGetSpatializer(lua_State *L) {
+  lua_pushstring(L, lovrAudioGetSpatializer());
   return 1;
 }
 
 static int l_lovrAudioNewSource(lua_State* L) {
   Source* source = NULL;
   SoundData* soundData = luax_totype(L, 1, SoundData);
-  AudioStream* stream = luax_totype(L, 1, AudioStream);
-  bool isStatic = soundData || luax_checkenum(L, 2, SourceType, NULL) == SOURCE_STATIC;
-
-  if (isStatic) {
-    if (soundData) {
-      source = lovrSourceCreateStatic(soundData);
-    } else {
-      if (stream) {
-        soundData = lovrSoundDataCreateFromAudioStream(stream);
-      } else {
-        Blob* blob = luax_readblob(L, 1, "Source");
-        soundData = lovrSoundDataCreateFromBlob(blob);
-        lovrRelease(Blob, blob);
-      }
-
-      lovrAssert(soundData, "Could not create static Source");
-      source = lovrSourceCreateStatic(soundData);
-      lovrRelease(SoundData, soundData);
-    }
+
+  bool spatial = true;
+  if (lua_istable(L, 2)) {
+    lua_getfield(L, 2, "spatial");
+    spatial = lua_isnil(L, -1) || lua_toboolean(L, -1);
+    lua_pop(L, 1);
+  }
+
+  if (soundData) {
```

Comment:
This if is really awkward to me

- Parsing flags should happen after resolving SoundData, this if should go earlier
- Ideally lovrSourceCreate would only be called in a single branch